lib/Etsy/StatsD.pm

```diff
@@ -20,20 +46,65 @@ Etsy::StatsD - Object-Oriented Client for Etsy's StatsD Server
 
 Create a new instance.
 
+=over
+
+=item HOST
+
+Default is localhost.  It may be in the form of '<host>:<port>', in which case,
+the port specified will be used instead of the PORT argument.  The argument may
+also be an array reference of strings in the form of "<host>" or
+"<host>:<port>".  If the port is not specified, the default port specified by
+the PORT argument will be used.
+
+=item PORT
+
+Default is 8125.  Will be used as the default port for any HOST argument not explicitly defining port.
+
+=item SAMPLE_RATE
+
+Default is undefined, or no sampling performed.  Specify a rate as a decimal between 0 and 1 to enable
+sampling. e.g. 0.5 for 50%.
+
+=back
+
 =cut
 
 sub new {
 	my ( $class, $host, $port, $sample_rate ) = @_;
 	$host = 'localhost' unless defined $host;
 	$port = 8125        unless defined $port;
 
-	my $sock = new IO::Socket::INET(
-		PeerAddr => $host,
-		PeerPort => $port,
-		Proto    => 'udp',
-	) or croak "Failed to initialize socket: $!";
-
-	bless { socket => $sock, sample_rate => $sample_rate }, $class;
+    # Handle multiple connections and
+    #  allow different ports to be specified
+    #  in the form of "<host>:<port>"
+    my @connections = ();
+    if( ref $host eq 'ARRAY' ) {
+        foreach my $addr ( @{ $host } ) {
+            my ($addr_host,$addr_port) = split /:/, $addr;
+            $addr_port //= $port;
+            push @connections, [ $addr_host, $addr_port ];
+        }
+    }
+    else {
+        my ($addr_host,$addr_port) = split /:/, $host;
+        $addr_port //= $port;
```

Comment:
In the past I've avoided the //= operator, since they were still some very old perls I needed to support.  I'm not sure if anyone is still using 5.8, and of course it would be less of an issue if this functionality is moved to a separate class.